### PR TITLE
added the requested stricter compiler settings; made appropriate adus…

### PR DESCRIPTION
**Pull request by 9Fingered**
Added the requested stricter compiler settings to ~/linux/system.mk.

Adjusted generated code as needed.

Adjusted test code as needed.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,12 +2,12 @@ cmake_minimum_required(VERSION 3.1...3.26)
 
 project(fsmlang VERSION 1.0 LANGUAGES C CXX)
 
-set(CMAKE_C_STANDARD 23)
+set(CMAKE_C_STANDARD 99)
```
Comment (Jasinsky) [inline CMakeLists.txt:5]: Why downgraded here?

File: CMakeLists.txt [hunk 2]
```diff
@@ -2,12 +2,12 @@ cmake_minimum_required(VERSION 3.1...3.26)
...
 set(CMAKE_C_STANDARD_REQUIRED ON)
 set(CMAKE_C_EXTENSIONS OFF)
 
 find_package(FLEX 2.6 REQUIRED)
-find_package(BISON 3.7 REQUIRED)
+find_package(BISON 3.5 REQUIRED)
```
Comment (Jasinsky) [inline CMakeLists.txt:10]: Why downgraded here?

Comment (9Fingered) [reply CMakeLists.txt:10]: Inadvertent - my installation needs it (can't get wsl linux to upgrade); I will restore.

File: CMakeLists.txt (hunk 1 above)
Comment (9Fingered) [reply CMakeLists.txt:5]: Inadvertent - my installation needs it (can't get wsl linux to upgrade); I will restore.

**Review by Jasinsky — APPROVED** [review]

**Pull request merged**